File: iOS/testReg/testReg/ViewController.m
//
//  ViewController.m
//  testReg
//
//  Created by 疯哥 on 8/18/16.
//  Copyright © 2016 疯哥. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    NSString *re = @"(?<=@)((?:[A-Za-z0-9]+(?:[\\-|\\.][A-Za-z0-9]+)*)+\\.[A-Za-z]{2,6})$";
    NSRange range = [email rangeOfString:re options:NSRegularExpressionSearch];
    if (range.location != NSNotFound) {
        return [email substringWithRange:range];
    } else {
      
    }
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: iOS/closure/closure/ViewController.swift
//
//  ViewController.swift
//  closure
//
//  Created by 疯之提琴 on 6/2/15.
//  Copyright (c) 2015 InfoGeeker. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        
        
        var names = ["Swift", "Arial", "Soga", "Donary"]
        
        // 第一种方式：使用函数
        func backwards(firstString: String, secondString: String) -> Bool {
            return firstString > secondString // 升序排序
        }
        // 这里第二个参数，传了一个函数
        // reversed is equal to ["Swift", "Soga", "Donary", "Arial"]
        var reversed = sort(&names, backwards)
        print(reversed)
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

